Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -1066,9 +1067,10 @@ final class BuildPlanTests: XCTestCase {
     }
 
     func testTestModule() throws {
+        let defaultEntryPointName = try XCTUnwrap(SwiftTarget.testEntryPointNames.first)
         let fs = InMemoryFileSystem(emptyFiles:
             "/Pkg/Sources/Foo/foo.swift",
-            "/Pkg/Tests/\(SwiftTarget.testManifestNames.first!).swift",
+            "/Pkg/Tests/\(defaultEntryPointName)",
```

Comment:
Note: this small change uncovered a pre-existing bug in the unit test: The prior logic appended an unintentional extra `.swift` file extension, resulting in `XCTMain.swift.swift`, and this caused it to _not_ correctly test the scenario of having a custom `XCTMain.swift` file as I think this test was intended to. Now that's fixed and I have restored the expected number of targets in the assertion below to 3 total (instead of 4, since it no longer synthesizes the test discovery target, it only synthesizes the test **entry point** target).